protocols/raft/src/main/java/io/atomix/protocols/raft/roles/LeaderRole.java

```diff
@@ -322,7 +322,13 @@ private boolean initializing() {
     }
 
     // If the member is already a known member of the cluster, complete the join successfully.
-    if (raft.getCluster().getMember(request.member().memberId()) != null) {
+    final MemberId memberId = request.member().memberId();
+    if (raft.getCluster().getMember(memberId) != null) {
+      final RaftMemberContext memberContext = raft.getCluster().getMemberState(memberId);
+      if (memberContext != null) {
+        // reset the failure count se we can immediately start appending again
+        memberContext.resetFailureCount();
```

Comment:
Is there more we'd want to reset here? The member context keeps track of a lot of things :thinking: I'm not that familiar with it, but maybe you have some idea.

Reply:
I already had a quick look and looks ok for now, but I'm currently investigating what else we can improve to reduce the time we spent in starting.